packages/integration/lib/src/views/feedroom/feedroom_list_screen.dart

```diff
@@ -0,0 +1,197 @@
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Choose FeedRoom"),
+        backgroundColor: LMFeedCore.theme.primaryColor,
+      ),
+      body: RefreshIndicator(
+        onRefresh: () async {
+          _feedRoomListBloc!.add(const LMFeedGetFeedRoomListEvent(offset: 1));
+          clearPagingController();
+        },
+        child: BlocConsumer<LMFeedRoomBloc, LMFeedRoomState>(
+          bloc: _feedRoomListBloc,
+          listener: (context, state) => updatePagingControllers(state),
+          buildWhen: (previous, current) {
+            if (current is LMFeedRoomListLoadingState &&
+                _feedRoomList.isNotEmpty) {
+              return false;
+            }
+            return true;
+          },
+          builder: ((context, state) {
+            if (state is LMFeedRoomListLoadedState) {
+              return FeedRoomListView(
+                pagingControllerFeedRoomList: _pagingControllerFeedRoomList,
+                feedRoomBloc: _feedRoomListBloc!,
+              );
+            } else if (state is LMFeedRoomListLoadingState) {
+              return getFeedRoomListLoadingView();
+            } else if (state is LMFeedRoomListErrorState) {
+              return getFeedRoomListErrorView(state.message);
+            } else if (state is LMFeedRoomListEmptyState) {
+              return getFeedRoomListEmptyView();
+            }
+            return Scaffold(
+              backgroundColor: LMFeedCore.theme.backgroundColor,
+              body: LMFeedLoader(),
+            );
+          }),
+        ),
+      ),
+    );
+  }
+}
+
+Widget getFeedRoomListEmptyView() {
+  return const Center(
+    child: Text("No feedrooms found"),
+  );
+}
+
+Widget getFeedRoomListErrorView(String message) {
+  return Center(
+    child: Text(message),
+  );
+}
+
+Widget getFeedRoomListLoadingView() {
+  return const LMFeedLoader();
+}
+
+class FeedRoomListView extends StatelessWidget {
+  final LMFeedRoomBloc feedRoomBloc;
+  final PagingController<int, LMFeedRoomViewData> pagingControllerFeedRoomList;
+  const FeedRoomListView({
+    super.key,
+    required this.pagingControllerFeedRoomList,
+    required this.feedRoomBloc,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        const SizedBox(height: 18),
+        Expanded(
+          child: PagedListView<int, LMFeedRoomViewData>(
+            pagingController: pagingControllerFeedRoomList,
+            builderDelegate: PagedChildBuilderDelegate<LMFeedRoomViewData>(
+                noMoreItemsIndicatorBuilder: (context) =>
+                    const SizedBox(height: 20),
+                noItemsFoundIndicatorBuilder: (context) => Scaffold(
+                      backgroundColor: LMFeedCore.theme.backgroundColor,
+                      body: LMFeedLoader(),
+                    ),
+                itemBuilder: (context, item, index) {
+                  return LMFeedTile(
```

Comment:
shouldn't feed room tile be customisable? Either using widget utility or using builder received in constructor